src/test/java/de/tum/in/www1/artemis/exam/ExerciseGroupIntegrationTest.java

```diff
@@ -163,25 +164,18 @@ void testDeleteExerciseGroup_asEditor() throws Exception {
     void importExerciseGroup_successfulWithExercisesIntoSameExam() throws Exception {
         Exam targetExam = examUtilService.addExamWithModellingAndTextAndFileUploadAndQuizAndEmptyGroup(course1);
 
-        final List<ExerciseGroup> listExpected = targetExam.getExerciseGroups();
+        final List<ExerciseGroup> exerciseGroupsBefore = targetExam.getExerciseGroups();
 
-        final List<ExerciseGroup> listReceived = request.postListWithResponseBody("/api/courses/" + course1.getId() + "/exams/" + targetExam.getId() + "/import-exercise-group",
-                listExpected, ExerciseGroup.class, HttpStatus.OK);
-        assertThat(listReceived).hasSize(9);
+        final List<ExerciseGroup> exerciseGroupsNow = request.postListWithResponseBody(
+                "/api/courses/" + course1.getId() + "/exams/" + targetExam.getId() + "/import-exercise-group", exerciseGroupsBefore, ExerciseGroup.class, HttpStatus.OK);
 
-        // We import into the same exam -> double the exercise groups
-        listExpected.addAll(listExpected);
+        assertThat(exerciseGroupsNow).hasSize(9).containsAll(exerciseGroupsBefore).allMatch(element -> element.getId() != null);
 
-        // The first 5 should be the old ones
-        for (int i = 0; i <= 4; i++) {
-            assertThat(listReceived.get(i)).isEqualTo(listExpected.get(i));
-        }
-        // The last 4
-        for (int i = 5; i <= 8; i++) {
-            assertThat(listReceived.get(i).getId()).isNotNull();
-            assertThat(listReceived.get(i).getId()).isNotEqualTo(listExpected.get(i).getId());
-            assertThat(listReceived.get(i).getTitle()).isEqualTo(listExpected.get(i).getTitle());
-            assertThat(listReceived.get(i).getIsMandatory()).isEqualTo(listExpected.get(i).getIsMandatory());
+        for (var exerciseGroup : exerciseGroupsBefore) {
+            assertThat(exerciseGroupsNow).filteredOn(element -> Objects.equals(element.getId(), exerciseGroup.getId())).hasSize(1);
+
+            assertThat(exerciseGroupsNow).filteredOn(element -> Objects.equals(element.getTitle(), exerciseGroup.getTitle()))
+                    .filteredOn(element -> Objects.equals(element.getIsMandatory(), exerciseGroup.getIsMandatory())).hasSizeGreaterThan(1);
```

Comment:
Why greater than 1 here? Are there groups with duplicated titles?

Reply:
As it is now, all groups seem to have the same title due to the ExamFactory helper function that is used. I think the test would make more sense if we change the helper function, which I will do now